### PR TITLE
Prevent trace strings from getting created when tracing is disabled

### PR DESCRIPTION
**Pull request by uni-cstar**
[issue#2493]( https://github.com/airbnb/lottie-android/issues/2493)

### CODE REVIEW

File: lottie/src/main/java/com/airbnb/lottie/animation/keyframe/BaseKeyframeAnimation.java [hunk 1]
```diff
@@ -112,8 +128,7 @@ protected float getInterpolatedCurrentKeyframeProgress() {
     Keyframe<K> keyframe = getCurrentKeyframe();
     // Keyframe should not be null here but there seems to be a Xiaomi Android 10 specific crash.
     // https://github.com/airbnb/lottie-android/issues/2050
-    // https://github.com/airbnb/lottie-android/issues/2483
-    if (keyframe == null || keyframe.isStatic() || keyframe.interpolator == null) {
+    if (keyframe == null || keyframe.isStatic()) {
```
Comment (gpeal) [inline lottie/src/main/java/com/airbnb/lottie/animation/keyframe/BaseKeyframeAnimation.java:131]: Let's leave this PR to a single task. Could you revert this?

**Review by gpeal — COMMENTED** [review]
I'd be surprised if this makes a meaningful impact with all of the ART optimizations and background GC improvements over the years but this seems like a reasonable optimization.

File: lottie/src/main/java/com/airbnb/lottie/animation/keyframe/BaseKeyframeAnimation.java (hunk 1 above)
Comment (uni-cstar) [reply lottie/src/main/java/com/airbnb/lottie/animation/keyframe/BaseKeyframeAnimation.java:131]: I found this issue was actually caused by virtual machine optimizations. I tracked memory allocations using the ‘Profiler’ tools from Android stuido and observed a significant number of StringBuilder instances being created at the points where beginSection and endSection methods were called. When I tested the code of my PR, these StringBuilder instances no longer appeared, and the memory stabilized without causing frequent garbage collection.
My test code is follow,I just provided an INFINITE LottieAnimationView in the layout, and didn't do anything else。

```
<com.airbnb.lottie.LottieAnimationView
                    android:id="@+id/assetView0"
                    android:layout_width="100dp"
                    android:layout_height="100dp"
                    app:lottie_autoPlay="true"
                    app:lottie_fileName="figure/data.json"
                    app:lottie_imageAssetsFolder="figure/images"
                    app:lottie_repeatCount="-1"/>
```

Comment (uni-cstar) [reply lottie/src/main/java/com/airbnb/lottie/animation/keyframe/BaseKeyframeAnimation.java:131]: I am Chinese and my English is not very good, so I may not fully understand your meaning or be able to respond clearly. I hope you can bear with me.

Comment (gpeal) [reply lottie/src/main/java/com/airbnb/lottie/animation/keyframe/BaseKeyframeAnimation.java:131]: This comment is specifically referring to the addition you made here for `keyframe.interpolator == null` which seems unrelated to the rest of the code in this PR.

**Comment by uni-cstar** [comment]
I fixed it.
















At 2024-04-13 04:20:18, "Gabriel Peal" ***@***.***> wrote:

@gpeal commented on this pull request.

In lottie/src/main/java/com/airbnb/lottie/animation/keyframe/BaseKeyframeAnimation.java:

> @@ -112,8 +128,7 @@ protected float getInterpolatedCurrentKeyframeProgress() {
     Keyframe<K> keyframe = getCurrentKeyframe();
     // Keyframe should not be null here but there seems to be a Xiaomi Android 10 specific crash.
     // https://github.com/airbnb/lottie-android/issues/2050
-    // https://github.com/airbnb/lottie-android/issues/2483
-    if (keyframe == null || keyframe.isStatic() || keyframe.interpolator == null) {
+    if (keyframe == null || keyframe.isStatic()) {


This comment is specifically referring to the addition you made here for keyframe.interpolator == null which seems unrelated to the rest of the code in this PR.

**Review by gpeal — APPROVED** [review]

**Pull request merged**